File: app/src/main/java/com/intfocus/yonghuitest/adapter/table/TableContentItemAdapter.java
package com.intfocus.yonghuitest.adapter.table;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.intfocus.yonghuitest.R;
import com.intfocus.yonghuitest.bean.table.Head;
import com.intfocus.yonghuitest.bean.table.MainData;
import com.intfocus.yonghuitest.util.Utils;

import java.util.List;

/**
 * Created by CANC on 2017/4/6.
 */

public class TableContentItemAdapter extends RecyclerView.Adapter<TableContentItemAdapter.TableHeadHolder> {
    private Context context;
    private List<MainData> mainData;
    private List<Head> heads;
    private final LayoutInflater inflater;
    private int rowHeight;//1行,2行,3行
    public ContentItemListener listener;


    public TableContentItemAdapter(Context context, List<Head> heads, List<MainData> mainData, int rowHeight, ContentItemListener listener) {
        this.context = context;
        this.mainData = mainData;
        this.inflater = LayoutInflater.from(context);
        this.rowHeight = rowHeight;
        this.listener = listener;
        this.heads = heads;
    }

    @Override
    public TableContentItemAdapter.TableHeadHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View contentView = inflater.inflate(R.layout.item_table_main_item, parent, false);
        return new TableContentItemAdapter.TableHeadHolder(contentView);
    }

    public void setData(List<Head> heads, List<MainData> mainData, int rowHeight) {
        this.heads = heads;
        this.mainData = mainData;
        this.rowHeight = rowHeight;
        this.notifyDataSetChanged();
    }

    @Override
    public void onBindViewHolder(TableContentItemAdapter.TableHeadHolder holder, final int position) {
        if (!heads.get(position).isShow() || heads.get(position).isKeyColumn) {
            holder.tvMain.setVisibility(View.GONE);
        } else {
            holder.tvMain.setVisibility(View.VISIBLE);
            holder.tvMain.setText(mainData.get(position).getValue());
            holder.tvMain.getLayoutParams().height = Utils.dpToPx(context, 50 * rowHeight);
            holder.tvMain.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    listener.ItemClick(position);
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return (mainData == null) ? 0 : mainData.size();
    }

    public class TableHeadHolder extends RecyclerView.ViewHolder {
        TextView tvMain;

        public TableHeadHolder(View itemView) {
            super(itemView);
            tvMain = (TextView) itemView.findViewById(R.id.tv_main);
        }
    }

    public interface ContentItemListener {
        void ItemClick(int position);
    }

}

File: app/src/main/java/com/intfocus/yh_android/subject/table/TableContentItemAdapter.kt
package com.intfocus.yh_android.subject.table

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.intfocus.yh_android.R
import com.intfocus.yh_android.bean.table.Head
import com.intfocus.yh_android.bean.table.MainData
import com.intfocus.yh_android.util.Utils

import android.text.TextUtils.TruncateAt.MIDDLE

/**
 * Created by CANC on 2017/4/6.
 */

class TableContentItemAdapter(private val context: Context, private var heads: List<Head>?, private var mainData: List<MainData>?, private var rowHeight: Int//1行,2行,3行
                              , var listener: TableContentItemAdapter.ContentItemListener) : RecyclerView.Adapter<TableContentItemAdapter.TableHeadHolder>() {
    var inflater = LayoutInflater.from(context)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TableContentItemAdapter.TableHeadHolder {
        val contentView = inflater.inflate(R.layout.item_table_main_item, parent, false)
        return TableContentItemAdapter.TableHeadHolder(contentView)
    }

    fun setData(heads: List<Head>, mainData: List<MainData>, rowHeight: Int) {
        this.heads = heads
        this.mainData = mainData
        this.rowHeight = rowHeight
        this.notifyDataSetChanged()
    }

    override fun onBindViewHolder(holder: TableContentItemAdapter.TableHeadHolder, position: Int) {
        if (!heads!![position].isShow || heads!![position].isKeyColumn) {
            holder.tvMain.visibility = View.GONE
        } else {
            holder.tvMain.text = mainData!![position].value
            holder.tvMain.ellipsize = MIDDLE
            holder.tvMain.maxLines = rowHeight
            holder.tvMain.visibility = View.VISIBLE
            holder.tvMain.layoutParams.height = Utils.dpToPx(context, (25 * rowHeight).toFloat())
            holder.tvMain.setOnClickListener { listener.itemClick(position) }
        }
    }

    override fun getItemCount(): Int {
        return if (mainData == null) 0 else mainData!!.size
    }

    class TableHeadHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvMain = itemView.findViewById(R.id.tv_main) as TextView
    }

    interface ContentItemListener {
    fun itemClick(position: Int)
}
}
